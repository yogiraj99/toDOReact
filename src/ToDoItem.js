import React, {Component} from 'react';

class ToDoItem extends Component {
  constructor(props) {
    super(props);
    this.state = props.state;
    this.toggleStatus = this.toggleStatus.bind(this);
    this.getTask = this.getTask.bind(this);
    this.showTaskToEdit = this.showTaskToEdit.bind(this);
    this.editTask = this.editTask.bind(this);
    this.saveTask = this.saveTask.bind(this);
  }

  render() {
    return (
        <div className={"toDoItem"} id={"toDoItem" + this.state.id}>
          <h4 className="task" ref="taskHeader" onClick={this.toggleStatus}>{this.getTask()}</h4>
          <button onClick={this.showTaskToEdit} ref="editButton" className="editButton">edit</button>
          <input value={this.state.task} ref={"task"} name="task" className="taskInput" onChange={this.editTask}/>
          <button onClick={this.saveTask} ref="saveButton" className="saveButton">Save</button>
        </div>
    )
  }

  getTask() {
    if (this.state.status) {
      return <del>{this.state.task}</del>
    }
    return this.state.task;
  }

  toggleStatus() {
    let status = !this.state.status;
    this.setState({status: status});
    this.props.update({task: this.state.task.value, id: this.state.id, status: status});
    return this.state.status;
  }

  editTask() {
    let task = this.refs.task.value;
    this.setState({task: task});
    this.props.update({task: this.refs.task.value, id: this.state.id, status: this.state.status});
    return this.state.status;
  }

  showTaskToEdit() {
    this.refs.task.style.display = "block";
    this.refs.saveButton.style.display = "block";
    this.refs.taskHeader.style.display = "none";
    this.refs.editButton.style.display = "none";
  }

  saveTask() {
    this.setState({task: this.refs.task.value});
    this.refs.task.style.display = "none";
    this.refs.saveButton.style.display = "none";
    this.refs.taskHeader.style.display = "block";
    this.refs.editButton.style.display = "block";
  }
}

export default ToDoItem
