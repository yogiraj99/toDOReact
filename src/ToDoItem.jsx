import React, {Component} from 'react';

class ToDoItem extends Component {
  render() {
    return <div className={"toDoItem"}>
      <input type={"checkbox"} checked={this.props.state.status} readOnly={true}/>
      <p>{this.props.state.task}</p>
    </div>
  }
}

export default ToDoItem
