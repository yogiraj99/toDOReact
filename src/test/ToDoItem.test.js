import React from 'react';
import ToDoItem from '../ToDoItem';
import renderer from 'react-test-renderer';

test('Should have className as "toDoItem" and should contain task', () => {
  let state={task:"Wake up early", status:true};
  const component = renderer.create(
      <ToDoItem state={state}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should have checkbox unchecked when status given as false', () => {
  let state={task:"Wake up early", status:false};
  const component = renderer.create(
      <ToDoItem state={state}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test('Should have checkbox checked when status given as true', () => {
  let state={task:"Wake up early", status:true};
  const component = renderer.create(
      <ToDoItem state={state}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});