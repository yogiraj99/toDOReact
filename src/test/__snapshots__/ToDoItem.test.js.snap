// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should have checkbox checked when status given as true 1`] = `
<div
  className="toDoItem"
>
  <input
    checked={true}
    readOnly={true}
    type="checkbox"
  />
  <p>
    Wake up early
  </p>
</div>
`;

exports[`Should have checkbox unchecked when status given as false 1`] = `
<div
  className="toDoItem"
>
  <input
    checked={false}
    readOnly={true}
    type="checkbox"
  />
  <p>
    Wake up early
  </p>
</div>
`;

exports[`Should have className as "toDoItem" and should contain task 1`] = `
<div
  className="toDoItem"
>
  <input
    checked={true}
    readOnly={true}
    type="checkbox"
  />
  <p>
    Wake up early
  </p>
</div>
`;
